From 79cd0651e21d92593e92b30e063fbbe5b81a78fe Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Fri, 21 Jun 2019 14:14:23 -0500
Subject: [PATCH] CMakeLists: Remove mavx and apply correct cflags

Based on v0.5.0

CMake: remove mavx
Codec CMake: add ssse3 flag
Codec CMake: add sse4.1
SSSE3 CMake: add cflags for Clang too
SSSE3 CMake: add sse4.1 flag
SSE4.1 CMake: add cflags to clang too
AVX2 CMake: convert march=avx2 flag to mavx2
CMake: Move include(CTest) inside BUILD_TESTING

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 CMakeLists.txt                              | 5 ++---
 Source/Lib/Common/ASM_AVX2/CMakeLists.txt   | 2 +-
 Source/Lib/Common/ASM_SSE4_1/CMakeLists.txt | 2 +-
 Source/Lib/Common/ASM_SSSE3/CMakeLists.txt  | 4 ++--
 Source/Lib/Common/Codec/CMakeLists.txt      | 4 ++++
 5 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1855a5ba..c200fc98 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,7 @@ if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 endif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")

 if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-    set(CMAKE_C_FLAGS "-Wall -Wextra -Wformat -Wformat-security -mavx -fstack-protector-strong -fPIE -fPIC -D_FORTIFY_SOURCE=2 ")
+    set(CMAKE_C_FLAGS "-Wall -Wextra -Wformat -Wformat-security -fstack-protector-strong -fPIE -fPIC -D_FORTIFY_SOURCE=2 ")
     set(CMAKE_C_FLAGS_RELEASE "-O2")
     set(CMAKE_C_FLAGS_DEBUG "-O0 -g")
 	SET(CMAKE_EXE_LINKER_FLAGS "-z noexecstack -z relro -z now -pie ")
@@ -48,8 +48,6 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     set(CMAKE_ASM_NASM_FLAGS_DEBUG "")
 endif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")

-include(CTest)
-
 set(COVERAGE false CACHE BOOL "For use with coveralls and GTest")

 # Prepare for Coveralls
@@ -73,6 +71,7 @@ add_subdirectory (Source/App/EncApp)
 add_subdirectory (Source/Lib/Decoder)
 add_subdirectory (Source/App/DecApp)
 if (BUILD_TESTING)
+    include(CTest)
     add_subdirectory (test)
     add_subdirectory (third_party/googletest)
 endif()
diff --git a/Source/Lib/Common/ASM_AVX2/CMakeLists.txt b/Source/Lib/Common/ASM_AVX2/CMakeLists.txt
index fbc40618..db6a8e3a 100644
--- a/Source/Lib/Common/ASM_AVX2/CMakeLists.txt
+++ b/Source/Lib/Common/ASM_AVX2/CMakeLists.txt
@@ -21,7 +21,7 @@ if(UNIX)
     if("${CMAKE_C_COMPILER_ID}" STREQUAL "Intel")
         SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC -static-intel -w")
     else()
-        SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=core-avx2")
+        SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mavx2")
     endif()
 else()
     # Intel Windows (*Note - The Warning level /W0 should be made to /W4 at some point)
diff --git a/Source/Lib/Common/ASM_SSE4_1/CMakeLists.txt b/Source/Lib/Common/ASM_SSE4_1/CMakeLists.txt
index 50ff420f..65e51e9b 100644
--- a/Source/Lib/Common/ASM_SSE4_1/CMakeLists.txt
+++ b/Source/Lib/Common/ASM_SSE4_1/CMakeLists.txt
@@ -18,7 +18,7 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     SET(CMAKE_C_FLAGS "/MP")
 endif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")

-if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+if(CMAKE_C_COMPILER_ID MATCHES "GNU|Clang")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse4.1")
 endif()

diff --git a/Source/Lib/Common/ASM_SSSE3/CMakeLists.txt b/Source/Lib/Common/ASM_SSSE3/CMakeLists.txt
index 65a4ab50..c2c23d9b 100644
--- a/Source/Lib/Common/ASM_SSSE3/CMakeLists.txt
+++ b/Source/Lib/Common/ASM_SSSE3/CMakeLists.txt
@@ -18,8 +18,8 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     SET(CMAKE_C_FLAGS "/MP")
 endif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")

-if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mssse3")
+if(CMAKE_C_COMPILER_ID MATCHES "GNU|Clang")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mssse3 -msse4.1")
 endif()

 if(CMAKE_C_COMPILER_ID STREQUAL "Intel")
diff --git a/Source/Lib/Common/Codec/CMakeLists.txt b/Source/Lib/Common/Codec/CMakeLists.txt
index 786f0ed1..6c32c691 100644
--- a/Source/Lib/Common/Codec/CMakeLists.txt
+++ b/Source/Lib/Common/Codec/CMakeLists.txt
@@ -17,6 +17,10 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     SET(CMAKE_C_FLAGS "/MP")
 endif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")

+if(CMAKE_C_COMPILER_ID MATCHES "GNU|Clang")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mssse3 -msse4.1")
+endif()
+
 file(GLOB all_files
     "*.h"
     "*.c")
--
2.21.0

